servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -5035,17 +5035,25 @@ List<Policy> setupPolicyList(AthenzDomain domain, Boolean assertions, Boolean ve
                             .setName(policy.getName())
                             .setModified(policy.getModified())
                             .setVersion(policy.getVersion())
-                            .setActive(policy.getActive());
+                            .setActive(policy.getActive())
+                            .setTags(policy.getTags());
                     policies.add(newPolicy);
                 }
             }
         }
 
+        if (tagKey != null) {
+            policies = policies.stream()
+                    .filter(policy -> filterByTag(tagKey, tagValue, policy, Policy::getTags))
+                    .filter(policy -> (versions != null && versions) || policy.active)
```

Comment:
few things:
1. this one is better : `versions == Boolean.TRUE`
2. do not access directly to attributes, but `policy.getActive()`
3. I think in this case multiple filter is incorrect, but single one:
```
policy -> filterByTag(tagKey, tagValue, policy, Policy::getTags) && (versions == Boolean.TRUE || policy.getActive() == Boolean.TRUE )
```